src/engine/v3/V3Interpreter.v3

```diff
@@ -1108,6 +1109,91 @@ class V3Interpreter extends WasmStack {
 			I32X4_TRUNC_SAT_F64X2_U_ZERO => do_v_v(V3Eval.I32X4_TRUNC_SAT_F64X2_U_ZERO);
 			I8X16_SHUFFLE => doShuffle();
 			I8X16_SWIZZLE => doSwizzle();
+			// ext: typed-continuation
+			CONT_NEW => {
+				var cont_index = codeptr.read_uleb32();
+				var cont_decl = ContDecl.!(frame.func.instance.heaptypes[cont_index]);
+				var func = Function.!(Value.Ref.!(pop()).val);
+				if (func == null) trap(TrapReason.NULL_DEREF);
+
+				var cont = Continuation.new(cont_decl, V3Interpreter.new());
+				cont.stack.reset(func);
+				push(Value.Ref(cont));
+			}
+			CONT_BIND => {
+				var in_cont = ContDecl.!(frame.func.instance.heaptypes[codeptr.read_uleb32()]);
+				var out_cont = ContDecl.!(frame.func.instance.heaptypes[codeptr.read_uleb32()]);
+				var n_binds = in_cont.sig.params.length - out_cont.sig.params.length;
+
+				var cont = Continuation.!(Value.Ref.!(pop()).val);
+				if (cont == null) trap(TrapReason.NULL_DEREF);
+				// continuation can only be resumed/bound once
+				if (cont.used) trap(TrapReason.USED_CONTINUATION);
+
+				var new_cont = Continuation.new(cont.cont, cont.stack);
+				for (i < cont.bounded.length) new_cont.bounded.put(cont.bounded[i]);
+
+				var range = values.peekn(n_binds);
+				for (i < range.length) new_cont.bounded.put(range[i]);
+				push(Value.Ref(cont));
+
+				values.top -= n_binds;
+			}
+			RESUME => {
+				var cont_idx = codeptr.read_uleb32();
+				var cont_type = ContDecl.!(frame.func.instance.heaptypes[cont_idx]);
+				var handlers = codeptr.read_handlers();
+				
+				var cont = Continuation.!(Value.Ref.!(pop()).val);
+				if (cont == null) trap(TrapReason.NULL_DEREF);
+				if (cont.used) trap(TrapReason.USED_CONTINUATION);
+
+				cont.used = true;
+				cont.stack.bind(values.peekn(cont_type.sig.params.length));
+				var r = cont.stack.resume();
```

Comment:
This works for now, but leaves the Virgil frames of the calling stack on the Virgil stack. That means that a long chain of resumes could overflow the Virgil stack. Let's think about how to fix this. I think we'll need to refactor the loop that calls `runWasmCode` so that there's a guard around it to find handlers for suspends and switch to the right stack.

Reply:
Sorry I must've missed this. I think in our last meeting we discussed moving this part out of the WasmStack and manage the stack switching `V3InterpreterOnlyStrategy`. I will try to implement that.